### PR TITLE
DNM: mgr/dashboard: Add ceph version to login page

### PR DESCRIPTION
**Pull request by zmc**
This work was inspired by https://tracker.ceph.com/issues/24647, but needs feedback since this is my first contribution to the dashboard (and my first time working with Angular). Commits are not squashed or signed-off on yet, and commit messages contain some notes about things that might want to be changed before a potential merge.

I'd love any and all feedback. Thanks!

### CODE REVIEW

**Comment by zmc** [comment]
screenshot

**Comment by yuriw** [comment]
:+1:

**Comment by callithea** [comment]
Hi @zmc, thanks for your contribution! :)
Unfortunately make check (mgr-dashboard-frontend-unittests) failed:
``Lint errors found in the listed files.
Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.``

**Comment by LenzGr** [comment]
Hi Zack, thanks a lot for your contribution and welcome!

I just noticed that it was actually me who submitted [issue#24647](https://tracker.ceph.com/issues/24647) and suggested to print the release/version information on the login page...

However, I must admit I made a mistake here - it's usually not a good idea to reveal this kind of information to unauthenticated users as it is considered [information leakage](http://projects.webappsec.org/w/page/13246936/Information%20Leakage) that potential attackers could exploit:

> Software version numbers and verbose error messages (such as ASP.NET version numbers) are examples of improper server configurations [7]. This information is useful to an attacker by providing detailed insight as to the framework, languages, or pre-built functions being utilized by a web application.

I am therefore inclined to reject this PR, as it would add a potential security risk. I am very sorry for have wasted your time; I have updated the tracker issue accordingly.

To my knowledge, we already display this version information on the "About" page once you've logged into the dashboard.

But I wonder if there is some other improvement on that issue or some other dashboard-related task that you would be interested in contributing instead?

How about one of these, for example?

https://tracker.ceph.com/issues/35689
https://tracker.ceph.com/issues/35690

File: src/pybind/mgr/dashboard/controllers/public.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from . import ApiController, BaseController, Endpoint
+from .. import mgr
+
+
+@ApiController('/public', secure=False)
+class Public(BaseController):
+    @Endpoint('GET')
+    def version(self):
+        raw_version = mgr.version
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/controllers/public.py]: As Lenz mentioned the version is actually provided in the `/summary` endpoint and the parsing of the different elements is done on the Front-end side by means of a pipe in the html template (pipes are in `src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/`).

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.html [hunk 2]
```diff
@@ -1,73 +1,78 @@
...
-                 #username="ngModel"
-                 type="text"
-                 placeholder="Enter your username..."
-                 class="form-control"
-                 required
-                 autofocus>
-          <div class="help-block"
-               *ngIf="(loginForm.submitted || username.dirty) && username.invalid">Username is required</div>
-        </div>
+  <div class="row full-height">
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.html]: If you feel like refactoring the login screen layout, we have [this other ticker](http://tracker.ceph.com/issues/35689) pending.

**Comment by zmc** [comment]
@LenzGr @epuertat Thanks for the feedback. While I wish I'd known it had been decided this feature wasn't desired any longer, working on it did enable me to learn a lot. I'll find something new to work on. I will probably mostly look at the backend and frontend, as I've been writing Python for ages and also have been wanting to branch out a bit into frontend work.

@callithea Not sure how I missed running the linter - thanks for pointing that out!

**Pull request closed without merging**